libs/hdf-converters/src/converters-from-hdf/splunk/reverse-splunk-mapper.ts

```diff
@@ -343,66 +344,64 @@ export class FromHDFToSplunkMapper extends FromAnyBaseConverter {
+      const hostname = config.port
+        ? `${config.scheme}://${config.host}:${config.port}`
+        : `${config.scheme}://${config.host}:8089`;
+      this.axiosInstance.defaults.params['sourcetype'] = MAPPER_NAME;
+      this.axiosInstance.defaults.params['index'] = targetIndex.name;
+
       // Upload execution event
-      splunkData.reports.forEach((report) => {
-        targetIndex.submitEvent(
-          JSON.stringify(report),
-          {
-            sourcetype: MAPPER_NAME,
-            index: targetIndex.name
-          },
-          (err: any) => {
-            if (err) {
-              reject(err);
-            }
-            logger.verbose(
-              `Successfully uploaded execution for ${report.meta.filename}`
-            );
-          }
-        );
-      });
+      for (const report of splunkData.reports) {
+        this.axiosInstance
+          .post(`${hostname}/services/receivers/simple`, JSON.stringify(report))
+          .then(
+            () =>
+              logger.verbose(
+                `Successfully uploaded execution for ${report.meta.filename}`
+              ),
+            (error) => reject(this.handleSplunkError(error))
+          );
+      }
       // Upload profile event(s)
       // \r\n Is the default LINE_BREAKER for splunk, this is very poorly documented.
       // See https://docs.splunk.com/Documentation/StreamProcessor/standard/FunctionReference/LineBreak
-      targetIndex.submitEvent(
-        splunkData.profiles
-          .map((profile) => JSON.stringify(profile))
-          .join('\n'),
-        {
-          sourcetype: MAPPER_NAME,
-          index: targetIndex.name
-        },
-        (err: any) => {
-          if (err) {
-            reject(err);
-          }
-          logger.verbose(
-            `Successfully uploaded ${splunkData.profiles.length} profile layer(s)`
-          );
-        }
-      );
+      this.axiosInstance
+        .post(
+          `${hostname}/services/receivers/simple`,
+          splunkData.profiles
+            .map((profile) => JSON.stringify(profile))
+            .join('\n')
+        )
+        .then(
+          () =>
+            logger.verbose(
+              `Successfully uploaded ${splunkData.profiles.length} profile layer(s)`
+            ),
+          (error) => reject(this.handleSplunkError(error))
+        );
 
       // Upload control event(s)
-      _.chunk(splunkData.controls, UPLOAD_CHUNK_SIZE).forEach((chunk) => {
-        targetIndex.submitEvent(
-          chunk.map((control) => JSON.stringify(control)).join('\n'),
-          {
-            sourcetype: MAPPER_NAME,
-            index: targetIndex.name
-          },
-          (err: any) => {
-            if (err) {
-              reject(err);
-            }
-            logger.verbose(`Successfully uploaded ${chunk.length} control(s)`);
-            resolve();
-          }
-        );
-      });
+      for (const chunk of _.chunk(splunkData.controls, UPLOAD_CHUNK_SIZE)) {
```

Comment:
these promises potentially don't complete.  they need to be wrapped in a promise.all or something and awaited.

keep an eye out for similar situations where promises are just created in order to do work within that same function instead of being returned for some user to deal with.

Reply:
Changed as requested.